tests/unit/amo/components/TestCollection.js

```diff
@@ -0,0 +1,445 @@
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({
+      location: newLocation,
+      params: { slug: newSlug, user },
+    });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, fetchCollection({
+      errorHandlerId: errorHandler.id,
+      page,
+      slug: newSlug,
+      user,
+    }));
+  });
+
+  it('dispatches fetchCollectionPage when page has changed', () => {
+    const store = dispatchClientMetadata().store;
+    const fakeDispatch = sinon.spy(store, 'dispatch');
+
+    // We need a collection for this test case.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const page = 123;
+    const location = { query: {} };
+    const newLocation = { query: { page } };
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'some-user';
+
+    const wrapper = renderComponent({
+      errorHandler,
+      location,
+      params: { slug, user },
+      store,
+    });
+    fakeDispatch.reset();
+
+    // This will trigger the componentWillReceiveProps() method.
+    wrapper.setProps({ location: newLocation });
+
+    sinon.assert.callCount(fakeDispatch, 1);
+    sinon.assert.calledWith(fakeDispatch, fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page,
+      slug,
+      user,
+    }));
+  });
+
+  it('renders a collection', () => {
+    const store = dispatchClientMetadata().store;
+
+    const collectionAddons = createFakeCollectionAddons();
+    const collectionDetail = createFakeCollectionDetail();
+
+    store.dispatch(loadCollection({
+      addons: collectionAddons,
+      detail: collectionDetail,
+    }));
+
+    const wrapper = renderComponent({ store });
+    expect(wrapper.find('.Collection-wrapper')).toHaveLength(1);
+    expect(wrapper.find(AddonsCard)).toHaveLength(1);
+    expect(wrapper.find(Paginate)).toHaveLength(1);
+  });
+
+  it('renders loading indicator on add-ons when fetching next page', () => {
+    const store = dispatchClientMetadata().store;
+
+    const errorHandler = createStubErrorHandler();
+    const slug = 'collection-slug';
+    const user = 'user-id-or-name';
+
+    // User loads the collection page.
+    store.dispatch(loadCollection({
+      addons: createFakeCollectionAddons(),
+      detail: createFakeCollectionDetail(),
+    }));
+
+    const wrapper = renderComponent({
+      errorHandler,
+      params: { slug, user },
+      store,
+    });
+    expect(wrapper.find(AddonsCard)).toHaveProp('loading', false);
+
+    // User clicks on 'next' pagination link.
+    store.dispatch(fetchCollectionPage({
+      errorHandlerId: errorHandler.id,
+      page: 2,
+      slug,
+      user,
+    }));
+
+    wrapper.setProps(mapStateToProps(store.getState()));
```

Comment:
Similar to https://github.com/mozilla/addons-frontend/pull/3268/files#r142172324 I'm confused about why you need to call `setProps()` again too. When you call `store.dispatch()` it should automatically trigger the update, no? Do you need to call `wrapper.update()` maybe? Or does it work without `setProps()` if you introduce `lifecycleExperimental: true` ?

Reply:
I did not know about `lifecycleExperimental` to be honest, and having `experimental` in an option does not sound very robust.

My main (and only) argument in favor of `setProps()` is that it is very explicit.

Reply:
So without `setProps()` it does not do anything but it makes sense according to the `shallow` documentation. Calling `update()` does not do anything. Let's see if I can use `lifecycleExperimental` now.

Reply:
mmh, no success with `lifecycleExperimental` 😢 I tried to pass it as `shallowOptions` to `shallowUntilTarget` but it did not work, and could not make it work with a regular `shallow`.

Reply:
I left a comment to explain why `setProps` is needed here.

Reply:
Huh, that's weird how `lifecycleExperimental` doesn't fix it. It's fine as is, I just wish I knew it why it's not working :/

Reply:
Yeah I am not entirely done with this, I want to spend 30 more minutes on it before I give up.

Reply:
Nope, no luck. Follow-up at: #3317.